gobblin-cluster/src/main/java/org/apache/gobblin/cluster/GobblinTaskRunner.java

```diff
@@ -191,6 +179,17 @@ public GobblinTaskRunner(String applicationName,
 
     this.containerMetrics = buildContainerMetrics();
 
+    logger.info("GobblinTaskRunner({}): applicationName {}, helixInstanceName {}, applicationId {}, taskRunnerId {}, config {}, appWorkDir {}",
+        this.isTaskDriver? "taskDriver" : "worker",
+        applicationName,
+        helixInstanceName,
+        applicationId,
+        taskRunnerId,
+        config,
+        appWorkDirOptional);
+  }
+
+  private synchronized TaskRunnerSuiteBase initTaskStateModelFactory() throws ReflectiveOperationException {
```

Comment:
this method has both return value and side effect which should be [avoided](https://softwareengineering.stackexchange.com/questions/289948/origin-of-a-method-should-return-a-value-or-have-side-effects-but-not-both). Shall we consider divide it into two: get TaskRunnerSuite  and initTaskStateModelFactory(TaskRunnerSuiteBase) if you want to make the init as a method ?